std/assembly/internal/typedarray.ts

```diff
@@ -123,4 +123,29 @@ export abstract class TypedArray<T,V> {
       return this;
     }
   }
+
+  map<W extends TypedArray<T, V>>(callbackfn: (value: T, index: i32, array: W) => T): TypedArray<T, V> {
```

Comment:
Wondering about `W extends XY` here. This is not supported by the compiler yet (though it might be able to parse it). Does `TypedArray#map` even have/need it?

Reply:
@dcodeIO according to the specification map passes the typedarray as the third argument.

```ts
    var cast = changetype<W>(this);

    for (let index = 0; index < length; ++index) {
      var val = loadUnsafeWithOffset<T,T>(buffer, index, byteOffset);
      storeUnsafeWithOffset<T, T>(
        newbuff,
        index,
        // it must be "cast" here (and I wasn't sure what to do about it
        callbackfn(val, index, cast),
        byteOffset,
      );
    }
```